src/oop/special_class_method/medium/q7/MainTest.java

```diff
@@ -0,0 +1,205 @@
+
+        );
+    }
+
+
+    @ParameterizedTest
+    @MethodSource("intInputProvider2")
+    public void correctToStringMethod(int value1, int value2) throws Throwable {
+        Object[][] arguments1 = {
+                {value1, int.class}
+        };
+        Object[][] arguments2 = {
+                {value2, int.class}
+        };
+        double value = (value1 * 1 + value2 * 5) / 100.0;
+        String incorrectToStringMessage = "Your toString method does not return the correct String.";
+        Object testInstance = testClass.createInstance(null);
+        testClass.callMethod(setAttributeMethodName1, arguments1, testInstance);
+        testClass.callMethod(setAttributeMethodName2, arguments2, testInstance);
+        String expectedOutput = toStringMessage + value;
+        Object toStringOutput = testClass.callMethod("toString", testInstance);
+        _assertEquals(expectedOutput, toStringOutput, incorrectToStringMessage);
+    }
+
+    @ParameterizedTest
+    @MethodSource("intInputProvider")
+    public void correctGetSetMethods(int updatedValue) throws Throwable {
+        String errorMessage = "Your " + className + " class " + getAttributeMethodName1 +
+                " method does not return the updated value after calling the " + setAttributeMethodName1 + " method.";
+        Object testInstance = testClass.createInstance();
+        Object[][] setMethodArguments = {
+                {updatedValue, int.class}
+        };
+        testClass.callMethod(setAttributeMethodName1, setMethodArguments, testInstance);
+        Object getMethodOutput = testClass.callMethod(getAttributeMethodName1, testInstance);
+        _assertEquals(updatedValue, getMethodOutput, errorMessage);
+    }
+
+    @ParameterizedTest
+    @MethodSource("intInputProvider")
+    public void correctGetSetMethods2(int updatedValue) throws Throwable {
+        String errorMessage = "Your " + className + " class " + getAttributeMethodName2 +
+                " method does not return the updated value after calling the " + setAttributeMethodName2 + " method.";
+        Object testInstance = testClass.createInstance();
+        Object[][] setMethodArguments = {
+                {updatedValue, int.class}
+        };
+        testClass.callMethod(setAttributeMethodName2, setMethodArguments, testInstance);
+        Object getMethodOutput = testClass.callMethod(getAttributeMethodName2, testInstance);
+        _assertEquals(updatedValue, getMethodOutput, errorMessage);
+    }
+
+    @ParameterizedTest
+    @MethodSource("intInputProvider2")
+    public void methodsWorkingTogether(int value1, int value2) throws Throwable {
+        Object testInstance = testClass.createInstance();
+        Object[][] setMethodArguments = {
+                {value1, int.class}
+        };
+        Object setMethodOutput = testClass.callMethod(setAttributeMethodName1, setMethodArguments, testInstance);
+        _assertEquals(value1, testClass.getFieldValue(testInstance, attributeName1),
```

Comment:
this should be switched to using the get method instead of getFieldValue

Reply:
the error message will also need to have the placement of set method and get method reversed